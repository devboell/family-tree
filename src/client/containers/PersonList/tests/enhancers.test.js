import { print } from 'graphql'

import {
  PERSONS_QUERY,
  personsQueryProps,
} from '../enhancers'

describe('Editor enhancers', () => {
  it('PERSONS_QUERY has a known shape', () => {
    expect(print(PERSONS_QUERY)).toMatchSnapshot()
  })

  it('correctly destructures query props', () => {
    const queryProps = {
      data: {
        persons: [],
        loading: true,
        refetch: jest.fn(),
      },
    }
    expect(personsQueryProps(queryProps)).toEqual(queryProps.data)
  })
})
