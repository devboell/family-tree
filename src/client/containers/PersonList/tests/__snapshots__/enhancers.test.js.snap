// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor enhancers PERSONS_QUERY has a known shape 1`] = `
"query personsQuery {
  persons {
    id
    ...personInfo
  }
}

fragment personInfo on Person {
  name
  title
  birthDate
  deathDate
  gender
  isMonarch
  bornToId
  partnerIds
  partners {
    relationId
    partner {
      id
      name
    }
    children {
      name
    }
  }
}
"
`;
